src/python/pants/backend/helm/test/unittest.py

```diff
@@ -85,10 +87,12 @@ async def run_helm_unittest(
     if len(chart_targets) == 0:
         raise MissingUnitTestChartDependency(field_set.address)
 
-    chart, source_files = await MultiGet(
-        Get(HelmChart, HelmChartRequest, HelmChartRequest.from_target(chart_targets[0])),
+    chart_target = chart_targets[0]
+    chart, chart_root, test_files = await MultiGet(
+        Get(HelmChart, HelmChartRequest, HelmChartRequest.from_target(chart_target)),
+        Get(SourceRoot, HelmChartSourceRootRequest(chart_target[HelmChartMetaSourceField])),
```

Comment:
I don't think you want to use the `SourceRoot` type for this at all. I fear it will lead to confusion with the "real" SourceRoots, as configured by the relevant options, which is what that type is documented to represent. 

Maybe create a new `HelmChartRoot` type?